pkg/plugins/networkinterface/apinet/apinet.go

```diff
@@ -59,10 +60,10 @@ func (p *Plugin) Init(host providerhost.Host) error {
 	return nil
 }
 
-func onmetalIPsToAPInetIPs(ips []string) []net.IP {
-	res := make([]net.IP, len(ips))
+func ironcoreIPsToAPInetIPs(ips []string) []apinet.IP {
+	res := make([]apinet.IP, len(ips))
 	for i, ip := range ips {
-		res[i] = net.MustParseIP(ip)
+		res[i] = apinet.MustParseIP(ip)
```

Comment:
why MustParseIP? usage of panic and handler recovery looks more as error exception system. Proper way for golang is return err.

Reply:
Fair point, but I would resolve this in a seperate PR as I want to change as least amount of logic as possible here.